addons/mrp/models/mrp_production.py

```diff
@@ -1418,27 +1418,31 @@ def _cal_price(self, consumed_moves):
         return True
 
     def _post_inventory(self, cancel_backorder=False):
+        moves_not_to_do = self.move_raw_ids.filtered(lambda x: x.state == 'done')
+        moves_to_do = self.move_raw_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+        for move in moves_to_do.filtered(lambda m: m.product_qty == 0.0 and m.quantity_done > 0):
+            move.product_uom_qty = move.quantity_done
+        # MRP do not merge move, catch the result of _action_done in order
+        # to get extra moves.
+        moves_to_do = moves_to_do._action_done(cancel_backorder=cancel_backorder)
```

Comment:
it was already the case before, but no need to catch returned value as it's overridden right after